### PR TITLE
Adds feature picking in Batched3DModelTileContent with ModelExperimental

### PR DESCRIPTION
**Pull request by sanjeetsuhag**
This PR adds support for feature picking for `Batched3DTileModelContent` when using `ModelExperimental`.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @sanjeetsuhag!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/model-experimental-b3dm-picking/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by XYCaptain** [comment]
It works excellent!

File: Source/Scene/Batched3DModel3DTileContent.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ import Model from "./Model.js";
 import ModelAnimationLoop from "./ModelAnimationLoop.js";
```
Comment (ptrgags) [inline Source/Scene/Batched3DModel3DTileContent.js:22]: This looks reasonable for now given we're under a time crunch, but it would be better to transcode the content to a Gltf3DTileContent. I opened https://github.com/CesiumGS/cesium/issues/9836 to list details like this for transcoding 1.0 formats

File: Source/Scene/ModelExperimental/FeatureIdPipelineStage.js [hunk 2]
```diff
@@ -49,43 +49,52 @@ FeatureIdPipelineStage.process = function (
...
+    var featureIdAttributePrefix;
     var featureIdAttributeSetIndex;
 
-    // Check if the Feature ID attribute references an existing vertex attribute.
-    if (defined(featureIdAttribute.setIndex)) {
-      featureIdAttributeSetIndex = featureIdAttribute.setIndex;
+    // For 3D Tiles 1.0, the FEATURE_ID vertex attribute is present but the Feature ID attribute is not.
+    // The featureMetadata is owned by the Cesium3DTileContent for the legacy formats.
+    var content = model.content;
+    if (defined(content) && defined(content._featureMetadata)) {
```
Comment (ptrgags) [inline Source/Scene/ModelExperimental/FeatureIdPipelineStage.js:58]: This nested if is getting a little hard to follow, consider moving it to a new function?

File: Source/Scene/ModelExperimental/ModelExperimentalPrimitive.js [hunk 3]
```diff
@@ -126,7 +127,8 @@ function initialize(runtimePrimitive) {
 
   var hasFeatureIds =
     defined(primitive.featureIdAttributes[featureIdAttributeIndex]) ||
-    defined(primitive.featureIdTextures[featureIdTextureIndex]);
+    defined(primitive.featureIdTextures[featureIdTextureIndex]) ||
+    defined(content.featureMetadata);
```
Comment (ptrgags) [inline Source/Scene/ModelExperimental/ModelExperimentalPrimitive.js:131]: This check fails in unit tests if `content` is undefined. check that content exists first.

**Review by ptrgags — COMMENTED** [review]
@sanjeetsuhag looked over the code and investigated the CI errors. I'll try this out in a sandcastle next.

**Comment by ptrgags** [comment]
Looks to be working in this [local sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=7Vptb9rIFv4rs3wp0SYGAmlIGqKbEHaDliRVobuqSlQN9oDndvBY9hjCvep/v2de/Bo7QFtVulU/JATPmfN+nnOOlT4JabS0Bk8+CeiSeAKzPwgWUUBCi3h4xsgddwjLnqMeEkFE3ky9qddooCsU0qXPCHLIkiM+R+0bNKGMhGiuGSGf2p+pt0BrKlzk8hUJEPYcFBJGbIFmxMUrygPF7DqizJG0DhYY2TwKBAk3kuv9hz568Il3Iw98HoAi58gVwj9vNNbrdcvyNra14KtGSAVpOJwK0aAeFRQLuiJho+0czQxzy8cLMvVWOEArStagTg95ZI362hd/q2f1ac1W3/scjKYeCaa1Q/TfqYeQIEEAT94GfEUdEpzHF+0ADCb/8IA5E01SPzicel8OlKu0KCu0iUesBeMzYjnEF+6EhOJqAcShMLcKDh4TgYRLkLaGIRsvSYCV6khwxDj/jLBAd9hzsRDY05YZ6rcc/EG55Gm07GPpU4q9tjUP+PKGLAJCwro07Oi0YzVb3W6r3TzuNpvHne6hfNxpWq/PWq1Ou90+bXW6xy319PSkPfWkaRlpDwGVKWIEZnx6S7D0/FsqbPcdZ+yZ6OOWdXzaPT3rnnaPz9rd9omScQSP2532WfPkpNk9Pe681qKbVvO09frspHUGyp51O60To0nWxdpNVkiEDGhdRc4BX1NP6Xde9JDizFMDzkuMUjTEcyYB9sI5D5ZJ7O+wCOhTxxreDO4nw8mHNO4QwRHHjgqhTOI4C0MkZJUQoT1ovuTdpj/aNxN9qI2IApaIHXLvHQmhTGyiXHoVAtnQqUNwOu0k+XJ+8aGOqaoJCztO3chNdL2d3I2QrFGGNwhMRKHL17Ii42r2wK8I4rvkUUgkodZfPn4w13rI4XYk0cLUxIAR+Q1qyqGraS2jkx1Xl4V9qG6n74J36hlmkjbz1bIZDsN7qUMPTWszbH92Au5PawWyUGwYFBgNfa3QtOZxj1SQ+WmRTGt4FnIWiSraGReCLxVls4KEkbl4ieDjtOZz6gGMHJEVuCWc1h63qgixklGQZB3/qUo5cMci4BH4kTMeaBcxeKjoVXyHnsxbXaJ4xiONLXYUBKAJ2xhUJk4cbx3d5GlPQ6A5PEcgiswhfk5cPguoL6akn+cyWT7RcJg0Dg9KSVAB8D77t+wEa5faLnQJxqBLMAfqb8530lHlqaSe8SdkYp7Xe6AF5YpLP6snmf8nVB8YgyMmkAqhzaBxIRd6FYMeIWWsXQJeS8TSEHlcqP4GyoBD02tavPrz1jDooaQOAfa8sY9tMpDxNwTWAkrX8yNxZcvoKFg0qo4LNyYbn1ijwR+TT/3RsP+XRj8d3jm0Y+byiAhAeYRByzDyZb+EEGWOYhtC8BuasSgtaZQ26JRaArWXkii7rCqB0icZoTZ4cEFUCGPH2So7oXltCGN8XSK7QiCdo6xfALsFtGFo1OFYQFMf0VmAg41Fw3Gi0zjWpJ6FwQPwmU5l2WGrPCbPVRolBNfSVb1t8g3qOQtyQwRR4VTnKtlQgZ0VgVlQk6Flm6LN1ox1PXo/ePEWI94CBquebIutMspM8X58VIlStOpP5e3vZ5bit8Wu0fBu8PK1lwzTpKWW5ZpdwRLd8yQZ2s3YTB4pSz/qu0UfHz5/rDQ0zx8P5OeB0TCXcRk4ictQjchWzpZywAjzgDGPPPUH8LiTVXMHfIytS/hTdmClhs57pQdUbyp/jUPkBwSG8SgElHXpwmXwo4oYcJ6rEk6eah47pJoS9FZCKVbwmzQWeSTzUOOnWTuyMKkiCIf1WH1YSZx4XtM5h5CEhN9MLE0vquc4HiQGI7TjiKCJAwLXPfP1S8aYq7zLlDAAV65mJUTFqzCZoOR8Y/y+TfqMcdOoy2njySNWL56gMLQaqDEmp9RbIkODjlCZx6wN+h3kmOmhXIiZXUqvPxWvx3MfXMh5XHYyKClYGaG/TmvXw3s98xVFCvIk+tpBsjXDSTZlbuNMy7UOCLdysGwymEGNOuk8YKUZkc+p33q9lMhwymTFC0mc4/OY5gL8OtzWoO8e3o8Hn+4e/h5sq3yoWS1UTWWy9cpEWhKB1R4MLVq6QI5u1/zp25BhBAHuy276DcgQO2gXWKgG6u+JC/63gEJ2SktYHmyHgbF+gbFrdqaaVXnnYxP2gF6hmDKKVmuCV3rAKk7GEonMTK6HrtS7GXhPXVxRCaDYm6rakgo/57VbdT2W2JKWPSRF6UKyLcGe1WwmYiINVbwvECgi6svsSZ1jiO63g5tErRTd8vuGZcAxw62cLqNDiu+v5GsDufNdwNKM1N7bi19KHVGNBUdM00xrlxcNILt8FUOzqZXi/pN/sI82F0K+DazSwyxOE0kjlREz7mwuX6Hf4/vT2oUI4Ll7Ce3gogGfF8K5nNZSiu0tJMetAdfhV5DjkUq5eRhOJmh4s7codfHT8GZPeeOH9+/6g68RqG/uIbGhnQuf0tuXph1v70fpwijT9QsiDDarqh0ot8RZ0HtCAWhWvc1ZSYfb9oYhAxbZmi5gUvymofJdw25vG8Avae/VL0XStpZun/9PI7eE4EJry9xLxpsMApccJztZ9izn0Dcv3IZ7STCyM9FPM+r/GvS/ZtDP98KyaEmKYpC2MKVOyvLLjjvED98gymskxyLJtdxbEJtBItbjo2KjUEV6mJ5WFmtMkk6tJYyKb2A+DEajh3++w05TxNWfbp0pgdxnyZCdeAtnifuTg3KkLd77ETD7Izenb96bnvlnj2WpwsFlNfoVG9XWvagENF4EgspSz8BBeTo9x4IdO3hmIKxQrwydKrQ4eBb23Ra7HXAreXP8a6f7tdP9JDudkTji3oKKyCF7S2TxzX0lwqL2dQLNxT3l6cF1b2muuranrPj/acywXB8wRv2QU+dgb/mG1e1eanynRV3OWbXD2oUa3C9j/v+iS7mcy38HqVtWQ5AlLBiwvTdmEZghLDsMYwS7aGSvKuihTq/kH5wSJJhHjI3pf0iCQcWrBqPMFC7J3NalATfLsi4a8LX8puCczXBQ4Pw/).

I also note that the default material color is different for modelExperimental... I'll try to make a quick PR for that, should be a tiny change in one file to change the default values.

File: Source/Scene/ModelExperimental/FeatureIdPipelineStage.js (hunk 2 above)
Comment (lilleyse) [inline Source/Scene/ModelExperimental/FeatureIdPipelineStage.js:58]: Can this use the getter instead of the private variable?

```suggestion
    if (defined(content) && defined(content.featureMetadata)) {
```

**Comment by ptrgags** [comment]
@sanjeetsuhag also make sure to sync in `main` since the picking code was modified in https://github.com/CesiumGS/cesium/pull/9837

File: Source/Scene/ModelExperimental/ModelExperimental.js [hunk 4]
```diff
@@ -138,7 +138,12 @@ function createModelFeatureTables(model, featureMetadata) {
   return modelFeatureTables;
 }
 
-function selectFeatureTableId(components, model) {
+function selectFeatureTableId(components, model, content) {
+  // For 3D Tiles 1.0 formats, the feature table always has the "_batchTable" feature table.
+  if (defined(content.featureMetadata)) {
+    return "_batchTable";
```
Comment (lilleyse) [inline Source/Scene/ModelExperimental/ModelExperimental.js:144]: It would be less error prone if `"_batchTable"` was a static property of some class

File: Source/Scene/ModelExperimental/ModelExperimental.js [hunk 5]
```diff
@@ -190,10 +195,14 @@ function initialize(model) {
     .then(function (loader) {
       var components = loader.components;
       var content = model._content;
-      var featureMetadata = components.featureMetadata;
+
+      // For 3D Tiles 1.0 formats, the feature metadata is owned by the Cesium3DTileContent classes.
+      var featureMetadata = defined(content.featureMetadata)
```
Comment (lilleyse) [inline Source/Scene/ModelExperimental/ModelExperimental.js:200]: Similar to @ptrgags's comment below, make sure to check that `content` is defined first. This will break code that create a `ModelExperimental` by itself like the `buildings-metadata` sandcastle in https://github.com/CesiumGS/cesium/issues/9832. Thankfully this is caught by unit tests but only when run locally, so make sure to always do that before opening a PR.

File: Source/Scene/Batched3DModel3DTileContent.js (hunk 1 above)
Comment (lilleyse) [reply Source/Scene/Batched3DModel3DTileContent.js:22]: I definitely second this. More and more special cases are starting to emerge making it hard to keep track of the logic.

File: Source/Scene/ModelExperimental/FeatureIdPipelineStage.js (hunk 2 above)
Comment (sanjeetsuhag) [reply Source/Scene/ModelExperimental/FeatureIdPipelineStage.js:58]: Updated to use the getter

File: Source/Scene/ModelExperimental/ModelExperimental.js (hunk 5 above)
Comment (sanjeetsuhag) [reply Source/Scene/ModelExperimental/ModelExperimental.js:200]: Added the required checks.

**Comment by lilleyse** [comment]
The recent changes look good. @ptrgags all set for you to give this a final look.

File: Source/Scene/MetadataClass.js [hunk 6]
```diff
@@ -156,4 +156,12 @@ Object.defineProperties(MetadataClass.prototype, {
   },
 });
 
+/**
+ * The class name given to the metadata class when a batch
+ * table is loaded from 3D Tiles 1.0 formats.
+ *
+ * @private
+ */
+MetadataClass._batchTableClassName = "_batchTable";
```
Comment (ptrgags) [inline Source/Scene/MetadataClass.js:165]: Should this be `MetadataClass.BATCH_TABLE_CLASS_NAME` since it's a constant?

**Review by ptrgags — COMMENTED** [review]
@sanjeetsuhag one minor question about variable naming, but otherwise the changes look good.

File: Source/Scene/MetadataClass.js (hunk 6 above)
Comment (sanjeetsuhag) [reply Source/Scene/MetadataClass.js]: Updated

**Comment by ptrgags** [comment]
looks good, I'll merge once CI passes.

**Comment by ptrgags** [comment]
@sanjeetsuhag merged!
